prow/git/v2/client_factory.go

```diff
@@ -262,6 +262,7 @@ func (c *clientFactory) bootstrapClients(org, repo, dir string) (cacher, cloner,
 			logger:   logger,
 		},
 		interactor: interactor{
+			locking:  locking,
```

Comment:
To me the `locking` field feels odd (the "-ing" being somewhat rare for field names). How about just `lock`, or even better (more to the point), `need_lock`? I especially like `need_lock` because you can imply that it's the `interactor` that needs the lock, whereas  with `locking` it's not clear (from just looking at this point of the code in isolation) whether it's the interactor or something else that needs the lock.

IIRC this is not a true lock in the sense that we are trying to ensure mutual exclusion. IIRC we are just trying to leave behind a file on disk to mark whether or not we finished processing the git repo cleanly. So, the presence of this file means "Git process pending, do not trust". I think this context is worth writing as a comment for the lock/unlock functions introduced in this commit.